### PR TITLE
server: remove disable learner option

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
We don't need `DisableLearner` anymore. 

### What is changed and how it works?
This PR removes this unused option.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by Luffbee** [comment]
I don't think it's a good idea to remove a public option, because it breaks the compatibility.

If this option won't be used anymore, marking it as deprecated is a better way.
If this option won't work anymore, leaving it there and give a warning or error in log if somebody changed it is more elegant.

**Comment by disksing** [comment]
It looks to be a good idea to keep the option and do nothing but gives a warning :smile:

**Comment by rleungx** [comment]
The main purpose I did this is to reduce the output item of `config show`. Any better idea? @disksing @Luffbee

**Comment by disksing** [comment]
`config show` output is too noisy. I think it is another problem that need to get fixed, maybe we can handle it separately.

**Comment by rleungx** [comment]
@disksing PTAL.

File: tools/pd-ctl/README.md [hunk 1]
```diff
@@ -266,12 +266,6 @@ The configuration above is global. You can also tune the configuration by config
     config set high-space-ratio 0.5             // Set the threshold value of sufficient space to 0.5
     ```
 
-- `disable-raft-learner` is used to disable Raft learner. By default, PD uses Raft learner when adding replicas to reduce the risk of unavailability due to downtime or network failure.
```
Comment (lhy1024) [inline tools/pd-ctl/README.md]: Is it better to add comment to describe that the option has been deprecated in Readme?

Comment (rleungx) [reply tools/pd-ctl/README.md]: Do you mean leave it as before and just mark this option as deprecated?

Comment (lhy1024) [reply tools/pd-ctl/README.md]: yep

**Comment by disksing** [comment]
@rleungx LGTM. Please resolve conflict.

**Pull request merged**